src/storage/compressed_tile.cpp

```diff
@@ -0,0 +1,301 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// compressed_tile.cpp
+//
+// Identification: src/storage/compressed_tile.cpp
+//
+// Copyright (c) 2015-16, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include <cstdio>
+#include <sstream>
+
+#include "catalog/schema.h"
+#include "common/exception.h"
+#include "common/macros.h"
+#include "type/serializer.h"
+#include "type/types.h"
+#include "type/ephemeral_pool.h"
+#include "concurrency/transaction_manager_factory.h"
+#include "storage/storage_manager.h"
+#include "storage/tile.h"
+#include "storage/tile_group_header.h"
+#include "storage/tuple.h"
+#include "storage/tuple_iterator.h"
+#include "storage/compressed_tile.h"
+
+namespace peloton {
+namespace storage {
+
+bool CompareLessThanBool(type::Value left, type::Value right) {
+  return left.CompareLessThan(right);
+}
+
+CompressedTile::CompressedTile(BackendType backend_type,
+                               TileGroupHeader *tile_header,
+                               const catalog::Schema &tuple_schema,
+                               TileGroup *tile_group, int tuple_count)
+    : Tile(backend_type, tile_header, tuple_schema, tile_group, tuple_count) {
+  is_compressed = false;  // tile is currently uncompressed.
+  compressed_columns_count = 0;
+}
+
+CompressedTile::~CompressedTile() {}
+
+void CompressedTile::CompressTile(Tile *tile) {
+  auto allocated_tuple_count = tile->GetAllocatedTupleCount();
+  std::vector<type::Value> new_column_values(allocated_tuple_count);
+  std::vector<std::vector<type::Value>> new_columns(column_count);
+  std::vector<catalog::Column> columns;
+  auto tile_schema = tile->GetSchema();
+
+  for (oid_t i = 0; i < column_count; i++) {
+    LOG_TRACE("Compressing Column %s",
+              tile_schema->GetColumn(i).GetName().c_str());
+    switch (tile_schema->GetType(i)) {
```

Comment:
There are other integer types in peloton, such as `TINYINT` or `TIMESTAMP`, which can be found in `src/include/type/types.h` or `value_factory.h`. It might be better to include all the cases.